### PR TITLE
Enabling json-c features

### PR DESCRIPTION
**Pull request by concatime**
First, is it dangerous to enable `rdrand`, even on non-x86/64?
Second, how to enable `do_check()`?
Third, can I use `make_check_args` instead of `do_check()`?

### CODE REVIEW

**Comment by maxice8** [comment]
> First, is it dangerous to enable rdrand, even on non-x86/64?

no clue about that

> Second, how to enable do_check()?

run ./xbps-src with -Q or XBPS_CHECK_PKGS=defined on `etc/conf`

> Third, can I use make_check_args instead of do_check()?

if you are using a build_style that makes use of it:

- configure
- gnu-configure
- gnu-makefile
- perl-module
- meson
- cmake

File: srcpkgs/json-c/template [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+desc_option_threading="Enable code to support partly multi-threaded use"
+desc_option_rdrand="Enable RDRAND Hardware RNG Hash Seed generation on supported x86/x64 platforms"
+
 CFLAGS="-Wno-error"
 
 pre_configure() {
 	autoreconf -fi
 }
+
+do_check() {
```
Comment (maxice8) [inline srcpkgs/json-c/template]: you're using build_style gnu-configure it has a definition of do_check() that can use make_check_args

File: srcpkgs/json-c/template [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
+checkdepends="valgrind"
 short_desc="A JSON implementation in C"
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
 license="MIT"
 homepage="http://oss.metaparadigm.com/$pkgname"
 distfiles="https://s3.amazonaws.com/json-c_releases/releases/$pkgname-$version.tar.gz"
 checksum=b87e608d4d3f7bfdd36ef78d56d53c74e66ab278d318b71e6002a369d36f4873
 
+build_options="threading rdrand"
+build_options_default=${build_options}
```
Comment (maxice8) [inline srcpkgs/json-c/template]: please be explicit with these

File: srcpkgs/json-c/template [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
-revision=1
+revision=2
 build_style=gnu-configure
+configure_args="$(vopt_enable threading) $(vopt_enable rdrand)"
 hostmakedepends="automake libtool"
+checkdepends="valgrind"
 short_desc="A JSON implementation in C"
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
 license="MIT"
 homepage="http://oss.metaparadigm.com/$pkgname"
```
Comment (maxice8) [inline srcpkgs/json-c/template]: new homepage= https://json-c.github.io/json-c/

File: srcpkgs/json-c/template [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
 distfiles="https://s3.amazonaws.com/json-c_releases/releases/$pkgname-$version.tar.gz"
```
Comment (maxice8) [inline srcpkgs/json-c/template]: would be nice if switched to the github distfiles in accordance with the new homepage= 

if possible

File: srcpkgs/json-c/template [hunk 5]
```diff
@@ -1,21 +1,32 @@
-# Template build file for 'json-c'.
```
Comment (maxice8) [inline srcpkgs/json-c/template]: needs a proper header

`# Template file for 'json-c'`

File: srcpkgs/json-c/template [hunk 6]
```diff
@@ -1,21 +1,30 @@
 # Template build file for 'json-c'.
```
Comment (maxice8) [inline srcpkgs/json-c/template]: please change to the appropiate header

`# Template file for 'json-c'`

Comment (concatime) [reply srcpkgs/json-c/template]: Even the dot at the end?

**Comment by concatime** [comment]
And is there a way to enable `check` in TravisCI? This way I can verify if `rdrand` on non-x86/64 would not harm the result.

**Comment by maxice8** [comment]
On Wed, Aug 08, 2018 at 10:26:33PM -0700, Issam Maghni wrote:
> concatime commented on this pull request.
> 
> 
> 
> > @@ -1,21 +1,30 @@
>  # Template build file for 'json-c'.
> 
> Even the dot at the end?
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/1608#discussion_r208809579

no dot at the end, sorry for that

**Comment by maxice8** [comment]
On Wed, Aug 08, 2018 at 10:30:00PM -0700, Issam Maghni wrote:
> And is there a way to enable `check` in TravisCI? This way I can verify if `rdrand` on non-x86/64 would not harm the result.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/1608#issuecomment-411641326

sorry, no way in the CI, you can run in the Travis of your fork with XBPS_CHECK_PKGS defined as an environment variable.

check travis-ci.org to add the envvar to your fork

**Comment by concatime** [comment]
Using what you suggested, I have one job that failed pretty hard (21 fails / 0 success)
https://travis-ci.org/concatime/void-packages/builds/413906456

**Comment by maxice8** [comment]
On Wed, Aug 08, 2018 at 10:54:58PM -0700, Issam Maghni wrote:
> Using what you suggested, I have one job that failed pretty hard (21 fails / 0 success)
> https://travis-ci.org/concatime/void-packages/builds/413906456
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/1608#issuecomment-411645339

you need to take a look at the logs that are pointed out 

"""
See tests/test-suite.log
"""

File: srcpkgs/json-c/template [hunk 7]
```diff
@@ -1,21 +1,30 @@
...
 maintainer="Juan RP <xtraeme@voidlinux.eu>"
 license="MIT"
-homepage="http://oss.metaparadigm.com/$pkgname"
-distfiles="https://s3.amazonaws.com/json-c_releases/releases/$pkgname-$version.tar.gz"
+homepage="https://${pkgname}.github.io/${pkgname}/"
+distfiles="https://s3.amazonaws.com/${pkgname}_releases/releases/${pkgname}-${version}.tar.gz"
 checksum=b87e608d4d3f7bfdd36ef78d56d53c74e66ab278d318b71e6002a369d36f4873
 
+build_options="threading rdrand"
+build_options_default="threading rdrand"
```
Comment (maxice8) [inline srcpkgs/json-c/template]: if rdrand is considered to be dangerous on non x86_64|i686 platforms you can guard them under a case/esac structure

**Comment by concatime** [comment]
So, after some debugging, the failing job is **not** due to `rdrand`, nor `threading`.
Logs https://travis-ci.org/concatime/void-packages/jobs/414119039#L1027

**Pull request closed without merging**